File: Exercism_Solutiuon/Scala/Triangle.scala
case class Triangle(a :Double, b :Double, c :Double) {
  
  def correct: Boolean = a > 0 && b > 0 && c > 0 && (a + b > c) && (a + c > b) && (b + c > a)
  
  val sides = Seq(a, b, c)
  
  def equilateral : Boolean = {
    if(correct && sides.distinct.size == 1) true
    else false
  }
  
  def isosceles : Boolean = {
    if (correct && sides.distinct.size <= 2 ) true
    else false
  }
  
  def scalene : Boolean = {
    if (correct && sides.distinct.size == 3 ) true
    else false
  }
}

File: Exercism_Solutiuon/R/Triangle.R
triangle <- function(x, y, z) {
  isTriangle <- c(x + y < z,
                  x + z < y,
                  y + z < x,
                  x == 0,
                  y == 0,
                  z == 0)

  stopifnot("error" != any(isTriangle))
  
  
  unique_sides <- length(unique(c(x, y, z)))
  triangle_class <- switch(unique_sides, 
                           c("equilateral", "isosceles"),
                           "isosceles",
                           "scalene")
  
  structure(list(x = x, y = y, z = z), class = triangle_class)
}